script.js

```diff
@@ -0,0 +1,223 @@
+    const weatherIconImg2 = await data.list[16].weather[0].icon;
+    const weatherIconImg3 = await data.list[24].weather[0].icon;
+    const weatherIconImg4 = await data.list[30].weather[0].icon;
+    const weatherIconImg5 = await data.list[36].weather[0].icon;
+    const weatherIconArray = [
+        weatherIconImg1, weatherIconImg2, weatherIconImg3, weatherIconImg4, weatherIconImg5
+    ]
+    return weatherIconArray;
+}
+async function renderData(data) {
+    const imgId = await data.list[0].weather[0].icon;
+    const todayTemp = (await data.list[0].main.feels_like - 273).toFixed(2);
+    const tempDetail = await data.list[0].weather[0].description;
+    tempDetailDisplay.textContent = tempDetail;
+    todayTempDisplay.textContent = todayTemp;
+    const windSpeed = await data.list[0].wind.speed;
+    windSpeedDisplay.textContent = windSpeed;
+    const humidity = await data.list[0].main.humidity;
+    humidityDisplay.textContent = humidity;
+    inputRange.value = humidity;
+    const visibility = await data.list[0].visibility;
+    const convertedVisibility = (visibility * 0.000621371).toFixed(2);
+    visibiltyDisplay.textContent = convertedVisibility;
+    const airPressure = data.list[0].main.pressure;
+    airPressureDisplay.textContent = airPressure;
+    todayWeatherImg.setAttribute('src', `http://openweathermap.org/img/wn/${imgId}@2x.png`);
+    cityInfo.textContent = data.city.name;
+}
+async function renderFutureData(data) {
+    let weathericonsArray = await weatherIconsData(data);
+    let minTempArray = await MinTempData(data);
+    let maxTempArray = await maxTempData(data);
+    for (let i = 0; i <= 5; i++) {
+        minTemp[i].textContent = minTempArray[i];
+        maxTemp[i].textContent = maxTempArray[i];
+        futureCastImg[i].setAttribute('src', `http://openweathermap.org/img/wn/${weathericonsArray[i]}@2x.png`);
+    }
+}
+async function getWeatherData() {
+    const weatherInfo = async (location) => {
+        location = await location;
+        const longitude = await location.coords.longitude;
+        const latitude = await location.coords.latitude;
+        let data = await fetchData(latitude, longitude);
+        renderData(data);
+        celciusBtn.click();
+        renderFutureData(data);
+    }
+    let failure = (err) => {
+        console.log(err);
+    }
+    navigator.geolocation.getCurrentPosition(weatherInfo, failure);
+}
+
+async function searchLocation() {
+    try {
+        const userData = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${inputLocation.value}&APPID=3f48274a20bac99a1e52fb12652e7fdc`, { mode: 'cors' })
+        const jsonData = await userData.json();
+        const latitude = jsonData.coord.lat
+        const longitude = jsonData.coord.lon
+        let data = await fetchData(latitude, longitude);
+        renderData(data);
+        celciusBtn.click();
+        renderFutureData(data)
+    }
+    catch {
+        console.log('not found')
+    }
```

Comment:
It's better to show a toast notification to your end user than to just simply log the error to the console.